github/src/main/java/io/syndesis/github/GitHubServiceImpl.java

```diff
@@ -96,27 +94,23 @@ private Repository getRepository(String name) throws IOException {
         }
     }
 
-    private Repository createRepo(String name) throws IOException {
+    protected Repository createRepo(String name) throws IOException {
         Repository repo = new Repository();
         repo.setName(name);
         return repositoryService.createRepository(repo);
     }
 
     private void createOrUpdateFiles(Repository repo, String message, Map<String, byte[]> files) throws IOException {
-        for (Map.Entry<String, byte[]> entry : files.entrySet()) {
-            // Wait a bit to let GitHub catch up
-            // See http://stackoverflow.com/questions/19576601/github-api-issue-with-file-upload for details
-            sleep(1000L);
-            createOrUpdateFile(repo, message, entry.getKey(), entry.getValue());
+        GitWorkflow gitWorkflow = new GitWorkflow();
+        Repository repository = getRepository(repo.getName());
+        if (repository == null) {
+            repository = createRepo(repo.getName());
+            gitWorkflow.createFiles(repo.getHtmlUrl(), repo.getName(), message, files, new UsernamePasswordCredentialsProvider(Tokens.getAuthenticationToken(), "") );
```

Comment:
So the KeyCloak authentication token is used as username and an empty string as password. Could you please elaborate a bit how this finally work to add the GitHub OAuth token ?